src/cbuffer.js

```diff
@@ -16,17 +16,23 @@
  *    returns <code>-1</code> if <code>a < b</code>, <code>0</code> if <code>a == b</code>, <code>1</code> otherwise.
  * @param {boolean} unique enforce element uniqueness: when <code>true</code> replace existing element with a new
  *    one on conflict; when <code>false</code> keep both elements.
+ * @param {function} isSub returns <code>true</code> if <code>the argument</code> is meant to replace another object.
```

Comment:
This change seems to be very complex. Let's keep it for now, but I think it can be simplified by, for example, linking messages to each other instead of doing it at the CBuffer level.

Reply:
Yes, it is complex. The idea was to keep the higher level code with as little change as possible.

But I'd rather make simplifying changes now. Are you suggesting adding a link from the original msg (a field in the message object) to the replacement and handling it in `topic.messages()` and `topic.forEach()`?

That basically will remove the need to add the replacement messages to `CBuffer`.

```
orig: {  // original message
  seq: 10,
  content: "abc",
  *replacement: repl1*
}

repl1: {  // replacement
  seq: 25,
  content: "qwerty"
  head {
    replace: ":10"
  }
}
```

Reply:
Problem statement: `CBuffer` keeps messages sorted by `seq`. The replacement message breaks the sort order: its `seq` value is greater than the `seq` of the message it replaces. 

Options:
1. Keep both,  scanning the full list for replacement pairs (your solution).
2. Keep both. Maintain a separate list of replacement pars, scan only that instead of the full list.
3. Keep both, linking them to each other (and possibly keeping a separate list of links), deal with it at the higher level.
4. Keep one. Replace old content with the new content in the CBuffer at the old `seq` value. Deal with the gap in `seq` at the new message location by converting new message into a shim, like deleted messages shims. This would require modifications at a higher level.

The higher level code in general should be aware that the message was replaced. It's probably useful to show it in the UI as _edited_ or some such.

Reply:
Updated as discussed. Please, take a final look.